src/Components/DataHistoryExplorer/Internal/TimeSeriesViewer/Internal/TimeSeriesChart/Internal/ChartCommandBar/ChartCommandBar.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+    IChartCommandBarProps,
+    IChartCommandBarStyleProps,
+    IChartCommandBarStyles
+} from './ChartCommandBar.types';
+import { getStyles } from './ChartCommandBar.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    CommandBar,
+    ICommandBarItemProps
+} from '@fluentui/react';
+import { IDataHistoryChartOptions } from '../../../../../../../../Models/Types/Generated/3DScenesConfiguration-v1.0.0';
+import { QuickTimeSpanKey } from '../../../../../../../../Models/Constants/Enums';
+import {
+    AggregationTypeDropdownOptions,
+    QuickTimeSpans
+} from '../../../../../../../../Models/Constants/Constants';
+import produce from 'immer';
+import { useTranslation } from 'react-i18next';
+import {
+    getQuickTimeSpanKeyByValue,
+    getQuickTimeSpanOptions,
+    getYAxisTypeOptions
+} from '../../../../../../../../Models/SharedUtils/DataHistoryUtils';
+import { capitalizeFirstLetter } from '../../../../../../../../Models/Services/Utils';
+
+const getClassNames = classNamesFunction<
+    IChartCommandBarStyleProps,
+    IChartCommandBarStyles
+>();
+
+const ChartCommandBar: React.FC<IChartCommandBarProps> = (props) => {
+    const { defaultOptions, onChange, deeplink, styles } = props;
+
+    // state
+    const [chartOptions, setChartOptions] = useState<IDataHistoryChartOptions>(
+        defaultOptions || {
+            yAxisType: 'independent',
+            defaultQuickTimeSpanInMillis:
+                QuickTimeSpans[QuickTimeSpanKey.Last15Mins],
+            aggregationType: 'avg'
+        }
+    );
+
+    // hooks
+    const { t } = useTranslation();
+
+    // callbacks
+    const handleChartOptionChange = useCallback(
+        (
+            chartOption:
+                | 'aggregationType'
+                | 'yAxisType'
+                | 'defaultQuickTimeSpanInMillis',
+            value: any
+        ) => {
+            setChartOptions(
+                produce((draft) => {
+                    draft[chartOption as string] = value;
```

Comment:
I tried to make a single callback method to update the corresponding chart option. Otherwise I had to do separate ones like handleAggregationTypeChange, handleYAxisTypeChange, handleQuickTimeChange... where the only thing I do is updating the chartOption state variables one corresponding field with the value. that is why draft[optionKey: 'aggregationType' | 'yAxisType' | 'defaultQuickTimeSpanInMillis'] = value would be a better approach.

But since it was barking for draft type I had to assign it to string:
![image](https://user-images.githubusercontent.com/45217314/213811988-6e105e78-6d13-4e2c-af44-f924e7705321.png)

Does it make sense now or I will change it to switch statement and directly updating the corresponding fields?